### PR TITLE
New serverless pattern - EventBridge-Bedrock-S3-AOSS

### PR DESCRIPTION
**Pull request by rajavaid77**
*Issue #, if available:* [2382](https://github.com/aws-samples/serverless-patterns/issues/2382)

*Description of changes:*

Added a serverless pattern to automate syncing an S3 data source sources to knowledge base for Amazon Bedrock using EventBridge scheduler. An EventBridge schedule is configured to run at a regular rate (default of 5mins). The EventBridge uses the universal target to trigger the StartIngestionJob API on the Amazon Bedrock Agent service. The StartIngestionJob operation in turn would sync the data source so that it is re-indexed to the knowledge base. Syncing is incremental, so Amazon Bedrock only processes added, modified, or deleted documents since the last sync.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eventbridge-bedrock-s3-aoss/template.yaml
Comment (pputhran) [inline eventbridge-bedrock-s3-aoss/template.yaml:1]: Is this needed?

**Review by pputhran — CHANGES_REQUESTED** [review]
Running into an error on deployment.
BedrockServiceRoleAccessPolicyStack and OpenSearchServerlessStack got deployed successfully.

BedrockKnowledgebaseStack is failing with the folliwing error

Resource handler returned message: "bedrock-rag-jcrob6 already exists (Service: S3, Status Code: 0, Request ID: null)" (RequestToken: f84ae2ac-9ac4-ad30-44c1-4a2c0a9bb599, HandlerErrorCode: AlreadyExists)

File: eventbridge-bedrock-s3-aoss/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      "This sample project demonstrates how to use an EventBridge rule to periodically sync a Bedrock Knowledge Base data source to Amazon OpenSearch Serverless.",
+      "The sample project uses an EventBridge Scheduler that runs on a regular interval. The scheduler invokes has a universal target configured to trigger the StartIngestion API on Amazon Bedrock Agent service"
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/eventbridge-bedrock-s3-aoss",
+      "templateURL": "serverless-patterns/eventbridge-bedrock-s3-aoss",
+      "projectFolder": "eventbridge-lambda-bedrock",
+      "templateFile": "eventbridge-lambda-bedrock_stack.py"
```
[the remark this answers is not in the document]
Comment (rajavaid77) [reply eventbridge-bedrock-s3-aoss/example-pattern.json]: fixed. pls double check

File: s3-lambda-bedrock-aoss/README.md
[the remark this answers is not in the document]
Comment (rajavaid77) [reply s3-lambda-bedrock-aoss/README.md:1]: it was added by mistake. now removed. thanks for finding this..

File: apigw-lambda-bedrock-s3-cdk-python/README.md
[the remark this answers is not in the document]
Comment (rajavaid77) [reply apigw-lambda-bedrock-s3-cdk-python/README.md:1]: it was added by mistake. now removed. thanks for finding this..

File: eventbridge-bedrock-s3-aoss/template.yaml
Comment (rajavaid77) [reply eventbridge-bedrock-s3-aoss/template.yaml:1]: no. deleted

File: eventbridge-bedrock-s3-aoss/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+1. Review the CloudFormation template CDK generates for the included stacks using the following AWS CDK CLI command:
+
+    > [!NOTE]  
+    > Substitute the stack_id with one from the list in output of the `cdk list` command
+    ```
+    cdk synth <stack_id>
+    ```
+1. From the command line, use AWS CDK to deploy the AWS resources.
+    ```
+    cdk deploy --all
```
[the remark this answers is not in the document]
Comment (rajavaid77) [reply eventbridge-bedrock-s3-aoss/README.md:104]: fixed the errors.

[the remark this answers is not in the document]
Comment (pputhran) [reply eventbridge-bedrock-s3-aoss/README.md:104]: Unfortunately ran into another error with the new changes with BedrockKBStack

Resource handler returned message: "Supplied Policy document is breaching Cloudwatch Logs policy length limit. (Service: CloudWatchLogs, Status Code: 400, Request ID: 39ff2d15-9eca-423f-bf0c-bf61d8d5bd4e)" (RequestToken: 972f5e1f-da40-7437-4d9b-5d6c5f5fea2e, HandlerErrorCode: AccessDenied)

**Review by pputhran — CHANGES_REQUESTED** [review]
Unfortunately ran into another error with the new changes with BedrockKBStack

**Comment by rajavaid77** [comment]
> Unfortunately ran into another error with the new changes with BedrockKBStack

This appears to be specific to a situation in the AWS account used for deploying the stack. Fixed by adding an explicitly policy that should solve the issue around limits.

**Pull request closed without merging**